### PR TITLE
Updating new rule generator to use yeoman

### PR DESCRIPTION
**Pull request by scalvert**
Updates the new rule generator with the following changes:

- changes the script name to `yarn new:rule` (more descriptive)
- uses file templates to output content (these can be expanded and improved later if we want to add more generator options)
- correctly inserts and sorts `rules/index.js` with the new rule ID

Screen Shot 2021-02-23 at 9 24 17 AM

Additionally:

- fixes the `scripts/update-rules.js` script to not include Octane (it no longer has rules of its own)
- renames the `yarn update` script to `yarn update:readme`

### CODE REVIEW

File: .eslintrc.js
Comment (rwjblue) [inline .eslintrc.js:155]: We should remove this override and instead add `dev/` to `.npmignore`

File: dev/new-rule/index.js
Comment (rwjblue) [inline dev/new-rule/index.js:3]: What is `wrap-iife`?

Can you write it as:

```js
async function main() {
  let env = createEnv();

  env.register(require.resolve('./new-rule-generator'), 'ember-template-lint:new-rule');

  await env.run('ember-template-lint:new-rule');
}

main();
```

This seems fine (old habbits with a "main function" die hard I guess), and avoids the linting comment

File: dev/new-rule/templates/rules-index.ejs
Comment (rwjblue) [inline dev/new-rule/templates/rules-index.ejs:3]: Let's put a comment in here saying this file is generated, and also add a `yarn update:rule-index` script to update it without having to add a new rule.

Comment (rwjblue) [reply dev/new-rule/templates/rules-index.ejs:3]: Then you can also add a test to ensure that `yarn update:rule-index` results in "no changes".

**Review by rwjblue — CHANGES_REQUESTED** [review]

**Comment by rwjblue** [comment]
Can you also update `CONTRIBUTING.md` (for both `yarn new` change and `yarn update` change)?

File: dev/new-rule/templates/rules-index.ejs
Comment (scalvert) [reply dev/new-rule/templates/rules-index.ejs:3]: So I _could_ do this, but if you look at what `updateRulesIndex` is doing, it's using yeoman's generators to generate the file. If I change this to move away from yeoman's `fs`, it also opts us out of the yeoman runLoop, which governs when the file is updated/written. I don't think we want to do that, as it results in strange interleaving with the yeoman write process.

I'm happy to do a similar thing in a separate util. Up to you.

File: dev/new-rule/index.js
Comment (scalvert) [reply dev/new-rule/index.js:3]: Yep, makes sense.

File: .eslintrc.js
Comment (scalvert) [reply .eslintrc.js:155]: Done.

File: dev/new-rule/templates/rules-index.ejs
Comment (rwjblue) [reply dev/new-rule/templates/rules-index.ejs:3]: Chatted offline about this with @scalvert. Let's make an issue to track this follow on work, it isn't blocking this from landing (it's just another nice improvement IMHO).

**Review by rwjblue — APPROVED** [review]
Generally LGTM (once CI is happy).

Only remaining TODO (other than CI) is to add an issue so we remember the follow on infra work RE: regenerating `rules/index.js`.

File: package.json
Comment (bmish) [inline package.json:27]: extra colon?

Comment (rwjblue) [reply package.json:27]: 😹

Comment (scalvert) [reply package.json]: Yep Fixed.

**Pull request merged**
